### PR TITLE
presets relativize paths

### LINKED ISSUE #15772: [question] Packaging prebuilt projects with existing FooConfing.cmake and Generating Multiple CMake Presets
### What is your question?

Hello,

I apologize for my lack of English,
We use conan to package manage our projects that use cmake and are used in various operating systems (macos, linux, android) and architectures (arm64, x86_64, x86, armv7, armv6). We can only use Conan for already compiled binaries because we need complex build systems to compile projects and even have to use docker for armv6 because the GCC version we use is extremely old (4.2.3) and the toolchain can only run on x86 Linux.

While developing the projects, we wrote a script that loads these packages with conan for all architectures and operating systems and creates a CMakePresets.json containing separate presets for each configuration. The presets contain their own FOO_DIR cache values.

The problem is that we need to rewrite this script for each project, and we cannot use CMakeToolChain or CMakeDeps because the presets they create are for only one configuration.

I wonder if there is a way to force CMakeToolChain and CMakeDeps to create presets for multiple configurations?

Also we don't want to edit our CMake files for conan and we dont want to use conan as a build tool because cmake do this excellently and we are not comfortable to depend on conan entirely even if its self hosted. 
### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: Relativize paths in `CMakePresets` generation.
Docs: Omit

Close https://github.com/conan-io/conan/issues/15772

### CODE REVIEW

**Review by czoido — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]
LGTM!

File: conan/tools/cmake/presets.py [hunk 1]
```diff
@@ -28,6 +29,11 @@ class _CMakePresets:
     @staticmethod
     def generate(conanfile, toolchain_file, generator, cache_variables, preset_prefix, buildenv, runenv,
                  cmake_executable):
+        toolchain_file = os.path.abspath(os.path.join(conanfile.generators_folder, toolchain_file))
+        try:  # Make it relative to the build dir if possible
+            toolchain_file = os.path.relpath(toolchain_file, conanfile.build_folder)
```
Comment (traceon) [inline conan/tools/cmake/presets.py:34]: This results in (wrong) paths like `generators/conan_toolchain.cmake` while the correct relative path would be `Release/generators/conan_toolchain.cmake` (relative to build dir for release build type, for example). It causes errors down the road, when included by `CMakeUserPresets.json`.

Comment (memsharded) [reply conan/tools/cmake/presets.py:34]: Hi @traceon 

Could you please file a new issue, providing full details to reproduce the issue?
Thanks

Comment (traceon) [reply conan/tools/cmake/presets.py:34]: Opened #19036

**Pull request merged**